test/boilerplate_suggestors/advanced_props_and_state_class_migrator_test.dart

```diff
@@ -0,0 +1,2420 @@
+                'AbstractBlockProps': 'AbstractBlockProps',
+              });
+            });
+          });
+
+          group('that is abstract based on annotation only (edge case)', () {
+            test('with members of its own', () {
+              const input = r'''
+                  @PropsMixin()
+                  abstract class BreadcrumbPathPropsMixin {}
+                  
+                  @AbstractProps()
+                  class _$AbstractBreadcrumbPathProps extends AbstractGraphFormProps
+                      with
+                          BreadcrumbPathPropsMixin,
+                          // ignore: mixin_of_non_class, undefined_class
+                          $BreadcrumbPathPropsMixin {
+                    String foo;
+                    int bar;
+                  }
+                  
+                  @AbstractComponent2()
+                  class AbstractBreadcrumbPathComponent<T extends AbstractBreadcrumbPathProps> 
+                      extends UiComponent2<T> {}
+                ''';
+
+              testSuggestor(visitedClassNames: {
+                'BreadcrumbPathPropsMixin': 'BreadcrumbPathPropsMixin',
+                'AbstractGraphFormProps': 'AbstractGraphFormProps',
+              })(
+                expectedPatchCount: 7,
+                input: input,
+                expectedOutput: r'''
+                @PropsMixin()
+                abstract class BreadcrumbPathPropsMixin {}
+              
+                @AbstractProps()
+                mixin AbstractBreadcrumbPathPropsMixin on UiProps {
+                  String foo;
+                  int bar;
+                }
+
+                @AbstractProps()
+                abstract class AbstractBreadcrumbPathProps 
+                    implements
+                        AbstractGraphFormProps,
+                        AbstractBreadcrumbPathPropsMixin,
+                        BreadcrumbPathPropsMixin {}
+
+                @AbstractComponent2()
+                class AbstractBreadcrumbPathComponent<T extends AbstractBreadcrumbPathProps> 
+                    extends UiComponent2<T> {}
+              ''',
+              );
+
+              expect(converter.visitedNames, {
+                'BreadcrumbPathPropsMixin': 'BreadcrumbPathPropsMixin',
+                'AbstractGraphFormProps': 'AbstractGraphFormProps',
+                'AbstractBreadcrumbPathProps':
+                    'AbstractBreadcrumbPathPropsMixin',
+              });
+            });
+
+            test('with no members', () {
+              const input = r'''
+                  @PropsMixin()
+                  abstract class BreadcrumbPathPropsMixin {}
+                  
+                  @AbstractProps()
+                  class _$AbstractBreadcrumbPathProps extends AbstractGraphFormProps
+                      with
+                          BreadcrumbPathPropsMixin,
+                          // ignore: mixin_of_non_class, undefined_clas
```

Comment:
If the ignore comment is at the end of a line, it'll get applied to this line, and not the subsequent line.

So, in this case, `$UnconvertedMixin` would still have an analysis error.

We need this comment to be on its own line:
```suggestion
                with ${publicPropsClassName}Mixin, ConvertedMixin, UnconvertedMixin, 
                // ignore: mixin_of_non_class, undefined_class
                \$UnconvertedMixin;
```
Doesn't matter where the comment goes so long as it has one space of between it and the start of the line; dartfmt will handle it:
```dart
@Props()
class FooProps = UiProps with
 FooPropsMixinn,
 ConvertedMixin,
 UnconvertedMixin,
 // ignore: mixin_of_non_class, undefined_class
 $UnconvertedMixin;
```
Formatted
```dart
@Props()
class FooProps = UiProps
    with
        FooPropsMixinn,
        ConvertedMixin,
        UnconvertedMixin,
        // ignore: mixin_of_non_class, undefined_class
        $UnconvertedMixin;
```